### PR TITLE
MAGECLOUD-3382: Increase default SCD_THREADS to 3

### PR DESCRIPTION
**Pull request by oshmyheliuk**
### Description
Default value of SCD_THREADS is set to 4 if enough CPU threads.

### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-3382

### Manual testing scenarios

https://jira.corp.magento.com/browse/MAGECLOUD-44
https://jira.corp.magento.com/browse/MAGECLOUD-53

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/App/Container.php [hunk 1]
```diff
@@ -171,6 +171,7 @@ function () {
                 return $this->container->makeWith(ProcessComposite::class, [
                     'processes' => [
                         $this->container->make(BuildProcess\PreBuild::class),
+                        $this->container->make(DeployProcess\PreDeploy\SetProductionMode::class),
```
Comment (shiftedreality) [inline src/App/Container.php]: Let's move it upper to `Process` namespace

File: src/Test/Unit/Config/SchemaTest.php [hunk 2]
```diff
@@ -39,7 +39,7 @@ public function testGetDefaultsForBuild()
                 BuildInterface::VAR_SKIP_SCD => false,
                 BuildInterface::VAR_SCD_COMPRESSION_LEVEL => 6,
                 BuildInterface::VAR_SCD_COMPRESSION_TIMEOUT => 600,
-                BuildInterface::VAR_SCD_THREADS => 1,
+                BuildInterface::VAR_SCD_THREADS => -1,
```
Comment (shiftedreality) [inline src/Test/Unit/Config/SchemaTest.php:42]: Nice ❤️

File: src/Test/Unit/Util/CpuTest.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    /**
+     * @var LoggerInterface|MockObject
+     */
+    private $loggerMock;
+
+    /**
+     * @var Cpu
+     */
+    private $cpu;
```
Comment (shiftedreality) [inline src/Test/Unit/Util/CpuTest.php]: Target class better to keep in top of properties, those won't move with adding/deleting them later

**Review by shiftedreality — CHANGES_REQUESTED** [review]

File: src/App/Container.php (hunk 1 above)
Comment (oshmyheliuk) [reply src/App/Container.php]: I thought about creation `Process\Common` folder to put there all processes that can be on all stages. What do you think?

Comment (shiftedreality) [reply src/App/Container.php]: That's would be over-complication, I would keep in `Process` as parent

**Review by shiftedreality — APPROVED** [review]

**Review by michaelcasey316 — APPROVED** [review]

**Comment by mveeramneni** [comment]
QA Approved

**Pull request merged**